core/src/main/java/org/elasticsearch/index/query/MissingQueryBuilder.java

```diff
@@ -50,23 +55,42 @@
 
     private boolean existence = DEFAULT_EXISTENCE_VALUE;
 
-    static final MissingQueryBuilder PROTOTYPE = new MissingQueryBuilder(null);
+    static final MissingQueryBuilder PROTOTYPE = new MissingQueryBuilder();
 
-    public MissingQueryBuilder(String fieldPattern) {
+    /**
+     * Constructs a filter that returns documents with only null values or no value in the original field.
+     * @param fieldPattern the field to query
+     * @param nullValue should the missing filter automatically include fields with null value configured in the
+     * mappings. Defaults to <tt>false</tt>.
+     * @param existance should the missing filter include documents where the field doesn't exist in the docs.
+     * Defaults to <tt>true</tt>.
+     * @throws IllegalArgumentException when both <tt>existence</tt> and <tt>nullValue</tt> are set to false
+     */
+    public MissingQueryBuilder(String fieldPattern, @Nullable Boolean nullValue, @Nullable Boolean existence) {
+        if (Strings.isEmpty(fieldPattern)) {
+            throw new IllegalArgumentException("missing query must be provided with a [field]");
+        }
+        if (nullValue != null) {
+            this.nullValue = nullValue;
+        }
+        if (existence != null) {
```

Comment:
shall we instead take in two primitive booleans here? if they are optional maybe add a constructor that only accepts the fieldPattern, and leave this one that forces users to set them both at the same time?